gpiozero/output_devices.py

```diff
@@ -898,3 +898,313 @@ def stop(self):
+        :attr:`value`).
+
+    :param float min_angle:
+        Sets the minimum angle that the servo can rotate to. This defaults to
+        -90, but should be set to whatever you measure from your servo during
+        calibration.
+
+    :param float max_angle:
+        Sets the maximum angle that the servo can rotate to. This defaults to
+        90, but should be set to whatever you measure from your servo during
+        calibration.
+
+    :param float min_pulse_width:
+        The pulse width corresponding to the servo's minimum position. This
+        defaults to 1ms.
+
+    :param float max_pulse_width:
+        The pulse width corresponding to the servo's maximum position. This
+        defaults to 2ms.
+
+    :param float frame_width:
+        The length of time between servo control pulses measured in seconds.
+        This defaults to 20ms which is a common value for servos.
+    """
+    def __init__(
+            self, pin=None, initial_angle=0.0,
+            min_angle=-90, max_angle=90,
+            min_pulse_width=1/1000, max_pulse_width=2/1000,
+            frame_width=20/1000):
+        self._min_angle = min_angle
+        self._angular_range = max_angle - min_angle
+        initial_value = 2 * ((initial_angle - min_angle) / self._angular_range) - 1
+        super(AngularServo, self).__init__(
+            pin, initial_value, min_pulse_width, max_pulse_width, frame_width)
+
+    @property
+    def min_angle(self):
+        """
+        The minimum angle that the servo will rotate to when :meth:`min` is
+        called.
+        """
+        return self._min_angle
+
+    @property
+    def max_angle(self):
+        """
+        The maximum angle that the servo will rotate to when :meth:`max` is
+        called.
+        """
+        return self._min_angle + self._angular_range
+
+    @property
+    def angle(self):
+        """
+        The position of the servo as an angle measured in degrees. This will
+        only be accurate if *min_angle* and *max_angle* have been set
+        appropriately in the constructor.
+
+        This can also be the special value ``None`` indicating that the servo
+        is currently "uncontrolled", i.e. that no control signal is being sent.
+        Typically this means the servo's position remains unchanged, but that
+        it can be moved by hand.
+        """
+        result = self._get_value()
+        if result is None:
+            return None
+        else:
+            # NOTE: Why round(n, 12) here instead of 14? Angle ranges can be
+            # much larger than -1..1 so we need a little more rounding to
+            # smooth off the rough corners!
+            return round(
+                self._angular_range *
+                ((result - self._min_value) / self._value_range) +
+                self._min_angle, 12)
+
+    @angle.setter
+    def angle(self, value):
+        if value is None:
+            self.value = None
+        else:
+            self.value = (
```

Comment:
Might be worth checking that the angle is between `_min_angle` and `_max_angle` and throwing an appropriate exception, as otherwise setting an invalid angle will result in a confusing "Servo value must be between -1 and 1, or None" error.
"Servo angle must be between _X_ and _Y_, or None" would be more appropriate IMHO.